props/saber_fett263_buttons.h

```diff
@@ -285,6 +445,217 @@ EFFECT(faston); // for EFFECT_FAST_ON
+  bool thruston; // Thrust On ignition
+  bool stabon; // Stab On ignition
+  bool forcepush; // Force Push
+  int forcepushlen; // Force Push Length
+  int lockupdelay; // Lockup Delay (for Battle Mode)
+  bool twistoff; // Twist Off retraction
+  // disable PWR button for retraction, for use with "Power Lock" mode 
+  // to prevent button turning saber off
+  bool powerlock;
+  int clashdetect; // maximum Clash Strength to detect Clash during Battle Mode (0 ~ 10 range)
+  int maxclash; // maximum Clash Strength for Clash Sound and Detection works with CLASH_THRESHOLD_G to create range of Clash Strength (8 ~ 16 range)
+};
+
+#ifdef FETT263_SAVE_CHOREOGRAPHY
+// Rehearsal / Choreography
+class SavedRehearsal : public ConfigFile {
+public:
+  virtual ReadStatus Read(FileReader* f) override {
+    int file;
+    if (!f || !f->IsOpen()) return ReadStatus::READ_FAIL;
+    memset(clash_rec, 0, sizeof(clash_rec));
+    STDOUT.println("Reading choreo.ini");
+    for (size_t i = 0; i < NELEM(clash_rec); i++) {
+      char action[33];
+      f->skipwhite();
+      f->readVariable(action);
+      if (!strcmp(action, "clash")) {
+        if (SFX_clsh) {
+           file = SFX_clsh.get_min_file();
+        } else {
+           file = SFX_clash.get_min_file();
+        }
+        clash_rec[i].stance = STANCE_CLASH;
+        f->skipspace();
+        clash_rec[i].sound_number1 = (int)(f->readFloatValue() - file);
+        f->skipspace();
+        continue;
+      }
+      if (!strcmp(action, "lockup")) {
+        if (SFX_lock) {
+          file = SFX_lock.get_min_file();
+        } else {
+          file = SFX_lockup.get_min_file();
+        }
+        clash_rec[i].stance = STANCE_LOCKUP;
+        f->skipspace();
+        clash_rec[i].sound_number1 = (int)(f->readFloatValue() - SFX_bgnlock.get_min_file());
+        f->skipspace();
+        clash_rec[i].sound_number2 = (int)(f->readFloatValue() - file);
+        f->skipspace();
+        clash_rec[i].sound_number3 = (int)(f->readFloatValue() - SFX_endlock.get_min_file());
+        f->skipspace();
+        continue;
+      }
+      clash_rec[i].stance = STANCE_END;
+      return ReadStatus:: READ_END;
+      break;
+    }
+  }
+
+  void SetVariable(const char* variable, float v) override {
+  }
+
+  enum FormStance {
+    STANCE_END = 0,
+    STANCE_CLASH,
+    STANCE_LOCKUP,
+  };
+
+  struct FormEntry {
+    FormStance stance;
+    int sound_number1;
+    int sound_number2;
+    int sound_number3;
+  };
+
+  FormEntry clash_rec[32];
+};
+#endif
+
+#ifdef FETT263_EDIT_MODE_MENU
+// Edit Length 
+int length_edit_length = 0;
+
+// Edit Length Preview for Edit Mode
+// LengthEdit uses blade color and creates single white pixel at last pixel, black above last pixel
+template<class BASE, class LIGHTUP, class BLACK = BLACK>
+class LengthEdit {
+public:
+  void run(BladeBase* blade) {
+    base_.run(blade);
+    lightup_.run(blade);
+    black_.run(blade);
+    led_ = length_edit_length;
```

Comment:
led_ is not needed, just use length_edit_length directly, here and in getColor()

Reply:
removed, ran into a bug from an earlier edit so had to update the menu control